### PR TITLE
Fix Java class references in documentation pages (#581)

### PR DESCRIPTION
**Pull request by jlprat**
Issue: #581
Went through all Java documentation pages and I replaced all inline snippets I could see that were using Scala syntax instead of the Java one.

### CODE REVIEW

**Comment by jlprat** [comment]
I also fixed the table that was not rendered under the Java TestKit page

File: docs/src/main/paradox/java/http/http-model.md [hunk 1]
```diff
@@ -83,20 +83,20 @@ It wraps a plain `ByteString` and  represents a standard, unchunked entity with
 
 HttpEntityDefault
 : The general, unchunked HTTP/1.1 message entity.
-It has a known length and presents its data as a `Source[ByteString]` which can be only materialized once.
+It has a known length and presents its data as a `Source<ByteString, ?>` which can be only materialized once.
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/http-model.md]: uff, thanks!

Comment (jlprat) [inline docs/src/main/paradox/java/http/http-model.md]: I fixed all I could see, there might be a couple more that escaped my checks

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
Failure was https://github.com/akka/akka-http/issues/595

**Comment by ktoso** [comment]
Thanks!

**Pull request merged**